Sources/PortalSwift/Provider/PortalProvider.swift

```diff
@@ -229,22 +229,26 @@ public class PortalProvider {
     if payload.method == ETHRequestMethods.WalletSwitchEthereumChain.rawValue {
       let param = payload.params[0] as! [String: String]
 
-      if let chainId = UInt8(param["chainId"]!.replacingOccurrences(of: "0x", with: ""), radix: 16) {
-        do {
-          _ = try self.setChainId(value: Int(chainId), connect: connect)
-
-          let completionResult = RequestCompletionResult(
-            method: payload.method,
-            params: payload.params,
-            result: "null",
-            id: ""
-          )
+      if let chainIdString = param["chainId"]?.replacingOccurrences(of: "0x", with: "") {
+        if let chainId = Int(chainIdString, radix: 16) {
+          do {
+            _ = try self.setChainId(value: Int(chainId), connect: connect)
```

Comment:
did we need to cast this to an int again here?  its already released but just calling out

Reply:
haha yes 😂